builtin/mainmenu/dlg_create_world.lua

```diff
@@ -127,8 +133,23 @@ local function create_world_buttonhandler(this, fields)
 		this:delete()
 		return true
 	end
+	
+	if fields["mg_flags"] ~= nil then
+		local mg_flags_dlg = create_mg_flags_dlg()
+		mg_flags_dlg:set_parent(this)
+		this:hide()
+		mg_flags_dlg:show()
+		return true
+	end
+
+	local ddhandled = false
+	if fields["dd_mapgen"] then
+		core.setting_set("mg_name",fields["dd_mapgen"])
+		ddhandled = true
+		return true
+	end
 
-	return false
+	return ddhandled
```

Comment:
I see it. But you return true in the dropdown. And only the dropdown is not handled you need to return false. I think you don't need ddhandled.